### PR TITLE
Missions

### PR DESCRIPTION
**Pull request by zewdu444**
## Implement Mission section 

- [x] Create basic structure for Missions-setup 
- [x] Fetch missions-fetch data from API
- [x] Display missions -Lists render
- [x] Implement mission leaving -Actions
- [x]  Implement mission Joining-Actions
- [x] Switch badges for missions-conditional components
- [x] Display joined mission in my profile-filters

### CODE REVIEW

File: src/components/Missions.js [hunk 1]
```diff
@@ -1,8 +1,65 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import Table from 'react-bootstrap/Table';
+import Button from 'react-bootstrap/Button';
+import { useDispatch, useSelector } from 'react-redux';
+import Badge from 'react-bootstrap/Badge';
+import { fetchMissions, joinMission } from '../redux/missions/missionsSlice.js';
```
Comment (Donmark2k) [inline src/components/Missions.js]: Please, I suggest that you ensure that the name of the file tarry while fetching the function. Please kindly look into it and update it. it should be   import { fetchMissions, joinMission } from '../redux/missions/MissionsSlice.js';

File: src/redux/store.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { configureStore } from '@reduxjs/toolkit';
 import { createLogger } from 'redux-logger';
-import { rocketsSlice } from './rockets/RocketsSlice';
+import missionsReducer from './missions/missionsSlice.js';
```
Comment (Donmark2k) [inline src/redux/store.js]: Please, I suggest that you ensure that the name of the file tarry while fetching the function. Please kindly look into it and update it. it should be import missionsReducer from './missions/MissionsSlice.js';

**Review by Donmark2k — CHANGES_REQUESTED** [review]
### Hi @zewdu444 

### I'm chukwuemeka Ochuba your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7

## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/store.js (hunk 2 above)
Comment (zewdu444) [reply src/redux/store.js]: @Donmark2k thank you for the feedback file name has been corrected

**Review by Donmark2k — APPROVED** [review]
Hi @zewdu444 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: no linter errors 
:+1: good styling

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**